### PR TITLE
DBZ-4910 / DBZ-5022 / DBZ-5023

### PR DESCRIPTION
**Pull request by smiklosovic**

### CODE REVIEW

File: cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java [hunk 1]
```diff
@@ -337,7 +336,7 @@ private void handlePartitionDeletion(PartitionUpdate pu, OffsetPosition offsetPo
         if (!clusteringColumns.isEmpty()) {
             throw new CassandraConnectorSchemaException("Uh-oh... clustering key should not exist for partition deletion");
         }
-        Collection<ColumnMetadata> columns = tableMetadata.getColumns().values();
+        List<ColumnMetadata> columns = new ArrayList<>(tableMetadata.getColumns().values());
```
Comment (smiklosovic) [inline cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java:339]: @jpechane @vjuranek this is the bug I was talking about in JIRA.

File: cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3SchemaChangeListener.java [hunk 2]
```diff
@@ -44,6 +44,7 @@ public void onSessionReady(Session session) {
         List<TableMetadata> cdcEnabledTableMetadataList = getCdcEnabledTableMetadataList(session);
         for (com.datastax.oss.driver.api.core.metadata.schema.TableMetadata tm : cdcEnabledTableMetadataList) {
             schemaHolder.addOrUpdateTableSchema(new KeyspaceTable(tm), new KeyValueSchema(this.kafkaTopicPrefix, tm, this.sourceInfoStructMaker));
+            onKeyspaceCreated(session.getMetadata().getKeyspace(tm.getKeyspace().toString()).get());
```
Comment (vjuranek) [inline cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3SchemaChangeListener.java:47]: This seems to be yet another issue. Could you please create a Jira issue also for this bug and move it into separate commit.

Comment (vjuranek) [inline cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3SchemaChangeListener.java:47]: This seems to be yet another issue. Could you please create a Jira issue also for this bug and move it into separate commit. Btw. the same issue seems to be in the `Cassandra4SchemaChangeListener`.

**Review by vjuranek — COMMENTED** [review]
Overall LGTM, but I'd suggest to separate bug fixes from refactoring and create corresponding Jira issue for the bug fixed here.

File: cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java (hunk 1 above)
Comment (smiklosovic) [reply cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java:339]: @vjuranek without this fix in the tests would fail so do I get it right we need to fix this first before this one can go in?

File: cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3SchemaChangeListener.java (hunk 2 above)
Comment (smiklosovic) [reply cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3SchemaChangeListener.java:47]: Ah ok.

File: cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java (hunk 1 above)
Comment (vjuranek) [reply cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java:339]: Yes. No problem to send it in one PR, but it would be nice to split it into separate commits (one fixing this issue, second one adding `onKeyspaceCreated()` and the last one for refactoring).

Comment (smiklosovic) [reply cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java:339]: @vjuranek done, it is building.

Comment (smiklosovic) [reply cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java:339]: I think I still need to create these separate jira tickets though :)) I ll do it.

Comment (smiklosovic) [reply cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java:339]: @vjuranek it is now fully done, DBZ-5022 and DBZ-5023.

**Review by vjuranek — APPROVED** [review]
Nice! Thanks a lot @smiklosovic !

**Comment by vjuranek** [comment]
@jpechane would you like to review it or I can merge it?

**Comment by jpechane** [comment]
@vjuranek You know more than me at this stage. Please feel free to do reviews and meging yourself. Just don't forget to resolve associated Jiras too. I suppose this is notheing that needs to be backported to 1.9 right?

**Comment by vjuranek** [comment]
@jpechane well, now I find out I haven't rights to merge it, so you have to do it anyway :-)

**Comment by jpechane** [comment]
@vjuranek Privileges granted :-)

**Comment by vjuranek** [comment]
ok, thanks @jpechane for you trust

**Comment by vjuranek** [comment]
and thanks @smiklosovic for you patches! Merged.
